### ISSUE TITLE
Bug about "player_example.cpp" in 02_write_a_player

### ISSUE THREAD

**Issue by great-luao** [sist-cs120/project-wiki#33]
It seems that the name of function "audioDeviceIOCallback" in Tester has been changed to "audioDeviceIOCallbackWithContext" in the new vision of JUCE. So in order to play the sound successfully, we need to update the function name and corr arguments.
![image](https://github.com/user-attachments/assets/df82caac-d1e3-4f75-93d5-12c7df08720c)
![image](https://github.com/user-attachments/assets/16d7c013-abed-454f-967f-be2d0cb5a495)

**Comment by SIST1C407** [comment]
Yes, this issue can be solved by either upgrading the code, or downgrading the JUCE library to v7.0.0
